packages/web-components/src/calendar/calendar.ts

```diff
@@ -0,0 +1,687 @@
+    super.disconnectedCallback();
+  }
+
+  public attributeChangedCallback(name: string, oldValue: string, newValue: string) {
+    // Sets focus on day grid cell when the month is updated on the day grid
+    if (name === 'month') {
+      if (this.navigatedDate.getMonth() + 1 != this.month || this.navigatedDate.getFullYear() != this.year) {
+        this.navigatedDate = new Date(`${this.month}-01-${this.year}`);
+      } else {
+        Updates.enqueue(() => {
+          const el = this.getNavigatedDayElement();
+          el.tabIndex = 3;
+          el.focus();
+        });
+      }
+    }
+
+    // Updates the secondaryPanelCells array and sets focus on the first cell
+    // on the secondary panel
+    if (name === 'monthpickeryear' || name === 'yearpickerdecade' || name === 'yearpickeropen') {
+      Updates.enqueue(() => {
+        this.secondaryPanelCells =
+          this.shadowRoot && Array.from(this.shadowRoot.querySelectorAll('.secondary-panel-cell-outer'));
+        if (this.secondaryPanelCells) {
+          this.secondaryPanelCells.forEach(cell => cell.setAttribute('tabindex', '-1'));
+          const focus = this.secondaryPanelCells[0] as HTMLElement;
+          focus.tabIndex = 6;
+        }
+      });
+    }
+  }
+
+  /**
+   * Gets data needed to render about a month picker year as well as the previous and next years
+   * @param year - year of the month picker
+   * @returns - an object with data about the current and 2 surrounding years
+   * @public
+   */
+  public getMonthPickerInfo(year: number = this.monthPickerYear): MonthPickerInfo {
+    return {
+      year,
+      previous: year - 1,
+      next: year + 1,
+    };
+  }
+
+  /**
+   * Gets data needed to render about a month picker year as well as the previous and next years
+   * @param decadeStart - the start of the decade on the year picker
+   * @returns - an object with data about the current and 2 surrounding decades
+   * @public
+   */
+  public getYearPickerInfo(decadeStart: number = this.yearPickerDecade): YearPickerInfo {
+    return {
+      decadeStart,
+      decadeEnd: decadeStart + NUM_YEARS_IN_DECADE - 1,
+      previousStart: decadeStart - NUM_YEARS_IN_DECADE,
+      nextStart: decadeStart + NUM_YEARS_IN_DECADE,
+    };
+  }
+
+  /**
+   * Checks if the calendar is on today's month and year
+   * @public
+   */
+  public isToday() {
+    const today: Date = new Date();
+
+    //when the month picker is not visible, this.monthPickerYear is always going to be the current year.
+    return (
+      this.month === today.getMonth() + 1 &&
+      this.year === today.getFullYear() &&
+      this.monthPickerYear === today.getFullYear()
+    );
+  }
+
+  /**
+   * Changes the month and year on the calendar
+   * @param month - the month to be switched to
+   * @param year - the year to be switched to
+   * @public
+   */
+  public handleSwitchMonth(month: number, year: number) {
+    this.selectedDates = '';
```

Comment:
We're not doing date ranges... yet. But when we do, how can we ensure that we preserve the selected dates when they span across months? We don't need to figure this out yet, but maybe adding a comment about a potential future approach might be helpful.

Reply:
After thinking about this a little more I think it's best to remove this line, since we want to remember the currently selected date even when the month changes. Thanks for pointing this out!!